src/pyfmi/simulation/assimulo_interface_fmi3.pyx

```diff
@@ -0,0 +1,611 @@
+        #     input_real_mask = []
+        #     self.input_other = []
+        #     input_other_mask = []
+
+        #     if isinstance(input_names,str):
+        #         input_names = [input_names]
+
+        #     for i,name in enumerate(input_names):
+        #         if self._model.get_variable_causality(name) != FMI2_INPUT:
+        #             raise FMUException("Variable '%s' is not an input. Only variables specified to be inputs are allowed."%name)
+
+        #         if self._model.get_variable_data_type(name) == FMI2_REAL:
+        #             self.input_real_value_refs.append(self._model.get_variable_valueref(name))
+        #             input_real_mask.append(i)
+        #         else:
+        #             self.input_other.append(name)
+        #             input_other_mask.append(i)
+
+        #     self.input_real_mask  = np.array(input_real_mask)
+        #     self.input_other_mask = np.array(input_other_mask)
+
+        #     self._input_activated = 1
+        # else:
+        #     self._input_activated = 0
+
+        # self.input = input
+
+    cpdef _set_input_values(self, double t):
+        pass
+        # if self._input_activated:
+        #     values = self.input[1].eval(t)[0,:]
+
+        #     if self.input_real_value_refs:
+        #         self._model.set_real(self.input_real_value_refs, values[self.input_real_mask])
+        #     if self.input_other:
+        #         self._model.set(self.input_other, values[self.input_other_mask])
+
+    cdef _update_model(self, double t, np.ndarray[double, ndim=1, mode="c"] y):
+        if self.model_me3_instance:
+            # Moving data to the model
+            self.model_me3._set_time(t)
+            # Check if there are any states
+            if self._f_nbr != 0:
+                self.model_me3._set_continuous_states_fmil(y)
+        else:
+            # Moving data to the model
+            self._model.time = t
+            # Check if there are any states
+            if self._f_nbr != 0:
+                self._model.continuous_states = y
+
+        # Sets the inputs, if any
+        self._set_input_values(t)
+
+    cdef int _compare(self, double t, np.ndarray[double, ndim=1, mode="c"] y):
+        cdef int res
+
+        if self.model_me3_instance:
+            if t != self.model_me3._get_time():
+                return 1
+
+            if self._f_nbr == 0:
+                return 0
+
+            self.model_me3._get_continuous_states_fmil(self._state_temp_1)
+            res = FMIL.memcmp(self._state_temp_1.data, y.data, self._f_nbr*sizeof(double))
+
+            if res == 0:
+                return 0
+
+            return 1
+        else:
+            return t != self._model.time or (not self._f_nbr == 0 and not (self._model.continuous_states == y).all())
+
+    def rhs(self, double t, np.ndarray[double, ndim=1, mode="c"] y, sw=None):
+        """
+        The rhs (right-hand-side) for an ODE problem.
+        """
```

Comment:
Lets put all of these short docstrings on one line, i.e. `""" foo. """`

Reply:
Fixed.